src/Symfony/Component/Validator/Constraints/ImageValidator.php

```diff
@@ -43,6 +44,22 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
+        if ($constraint->strict) {
+            if (!function_exists('imagecreatefromstring')) {
+                throw new RuntimeException('Strict image validation requires installed and enabled GD extension');
```

Comment:
I think the error should be triggered when the constraint is configured not when the validation happens.